pkg/cloudprovider/providers/vsphere/vclib/connection_test.go

```diff
@@ -94,7 +94,7 @@ func TestWithValidCaCert(t *testing.T) {
 	}
 
 	// Ignoring error here, because we only care about the TLS connection
-	connection.NewClient(context.Background())
+	_, _ = connection.NewClient(context.Background())
```

Comment:
does a linter complain about these?
it shouldn't.

Reply:
Yes it does, because the error return gets implicitly ignored. With this change, it gets explicitly ignored.

Why do you think it shouldn't complain about these?

Reply:
what is the exact linter message?

just tested with `golint` locally.
something in the lines of this does not cause an issue:
```
package main

import "errors"

func test(returnError bool) (string, error) {
    if returnError {
        return "", errors.New("some error")
    }
    return "hello world", nil
}

func main() {
    test(true)
    test(false)
}
```

> Why do you think it shouldn't complain about these?

it depends. the programmer has the freedom, to execute the logic without caring about the outcome of function - i.e. in a case where the function also performs extra work unrelated to the returned result. linters in C do complain about these, then again they also complain unused return values from `printf`.

Reply:
The linter I used to find these is errcheck: https://github.com/kisielk/errcheck

It basically complains every time a returned `error` is not explicitly handled, so for your example this is the result:

```
$ errcheck t.go 
t.go:13:6:	test(true)
t.go:14:6:	test(false)
```
>  it depends. the programmer has the freedom to execute the logic without caring about the outcome of function - i.e. in a case where the function also performs extra work unrelated to the returned result

Yes but the returned error may also indicate that during that extra work unrelated to the other returns an error was encountered. 
I personally find it a sensible requirement to explicitly ignore all errors, its the only safe way to avoid doing so by accident.

Reply:
since the PR is against `k/k`, to me it feels like unless `./hack/verify-golint.sh` (uses [golint](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-golint.sh#L33)) complains about the `_, _ =` case i don't think we should fix these based on a third party linter.

but the decision is obviously up to the maintainers, of course. :+1: